pkg/agent/discoverycache.go

```diff
@@ -85,8 +93,22 @@ func (j *discoveryCache) Status() error {
 	return fmt.Errorf("could not establish a connection to APIC to update the cache")
 }
 
-// Execute -
+// Execute - starts a loop to wait for the discovery cache to receive a signal to rebuild the cache
 func (j *discoveryCache) Execute() error {
+	for {
```

Comment:
For loop is not required here since select will be blocking in this case, so it can be removed

Reply:
I think the for is needed so that it can come back to waiting for a signal. Without the for loop it will wait here for one write from either stop or signal, and after that execute will end